File: Coders/DSA Concepts/bubble_sort.py
def bubbleSort(arr):
   n = len(arr)
   # Traverse through all array elements
   for i in range(n):
   # Last i elements are already in correct position
       for j in range(0, n-i-1):
          # Swap if the element found is greater than the next element
            if ar[j] > ar[j+1] :
                ar[j], ar[j+1] = ar[j+1], ar[j]
# Driver code to test above
ar = ['t','u','t','o','r','i','a','l']
bubbleSort(ar)
print ("Sorted array is:")
for i in range(len(ar)):
   print (ar[i])

File: Coders/DSA Concepts/BubbleSort.java
import java.util.*;
public class BubbleSort{
    public static  void bubbleSort(int arr[]){
        for (int turn =0;turn<arr.length-1;turn++){
            for (int j=0;j<arr.length-1-turn;j++){
                if (arr[j]>arr[j+1]){
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }
        }
    }
    public static void main(String args[]){
        int arr[] = {5,4,1,3,2};
        bubbleSort(arr);
        for (int i=0;i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }
        System.out.println();
    }   
}             